### PR TITLE
migrate IndexableBlockNumber to Head: has pk, backed by int64

### PR DESCRIPTION
**Pull request by dimroc**

### CODE REVIEW

File: store/migrations/migration1551895034/migrate.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	// db specific bytes -> hexadecimal conversion operation
+	conversion := "hex(hash)" // sqlite default
+	if orm.IsPostgres() {
+		conversion = "encode(hash::bytea, 'hex')"
+	}
+
+	tx := orm.DB.Begin()
+	err := tx.Exec(fmt.Sprintf(`
```
Comment (se3000) [inline store/migrations/migration1551895034/migrate.go]: 👍

File: store/orm/orm.go [hunk 2]
```diff
@@ -707,14 +707,14 @@ func (orm *ORM) CreateInitiator(initr *models.Initiator) error {
...
-func (orm *ORM) SaveHead(n *models.IndexableBlockNumber) error {
+func (orm *ORM) SaveHead(n *models.Head) error {
 	return orm.DB.Save(n).Error
 }
 
-// LastHead returns the last ordered IndexableBlockNumber.
-func (orm *ORM) LastHead() (*models.IndexableBlockNumber, error) {
-	number := &models.IndexableBlockNumber{}
-	err := orm.DB.Order("digits desc, number desc").First(number).Error
+// LastHead returns the last Head in descending order.
```
Comment (j16r) [inline store/orm/orm.go]: `returns the last Head in descending order` this doesn't make any sense, a single head has no order.

File: store/migrations/migration1551895034/old/models.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package old
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/chainlink/store/models"
+)
+
+// IndexableBlockNumber captured in old state for tests.
+type IndexableBlockNumber struct {
```
Comment (j16r) [inline store/migrations/migration1551895034/old/models.go]: Perhaps this should be private so people don't accidentally use it?

Comment (dimroc) [reply store/migrations/migration1551895034/old/models.go]: It's accessed outside the `old` package so needs to be public.

Comment (j16r) [reply store/migrations/migration1551895034/old/models.go]: It's only used in `store/migrations/migrate_test.go` perhaps it should go there as private?

File: store/orm/orm.go (hunk 2 above)
Comment (dimroc) [reply store/orm/orm.go]: Open to a suggestion.

How about
```
LastHead returns the most recent head entry.
```
?

**Pull request merged**